x-pack/plugins/ml/public/application/jobs/jobs_list/components/reset_job_modal/reset_job_modal.tsx

```diff
@@ -0,0 +1,151 @@
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiLoadingSpinner,
+  EuiText,
+} from '@elastic/eui';
+
+import { resetJobs } from '../utils';
+import { RESETTING_JOBS_REFRESH_INTERVAL_MS } from '../../../../../../common/constants/jobs_list';
+import type { MlSummaryJob } from '../../../../../../common/types/anomaly_detection_jobs';
+import { OpenJobsWarningCallout } from './open_jobs_warning_callout';
+
+type ShowFunc = (jobs: MlSummaryJob[]) => void;
+
+interface Props {
+  setShowFunction(showFunc: ShowFunc): void;
+  unsetShowFunction(): void;
+  refreshJobs(): void;
+}
+
+export const ResetJobModal: FC<Props> = ({ setShowFunction, unsetShowFunction, refreshJobs }) => {
+  const [resetting, setResetting] = useState(false);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [jobIds, setJobIds] = useState<string[]>([]);
+  const [jobs, setJobs] = useState<MlSummaryJob[]>([]);
+
+  useEffect(() => {
+    if (typeof setShowFunction === 'function') {
+      setShowFunction(showModal);
+    }
+    return () => {
+      if (typeof unsetShowFunction === 'function') {
+        unsetShowFunction();
+      }
+    };
+  }, []);
+
+  const showModal = useCallback((tempJobs: MlSummaryJob[]) => {
+    setJobIds(tempJobs.map(({ id }) => id));
+    setJobs(tempJobs);
+    setModalVisible(true);
+    setResetting(false);
+  }, []);
+
+  const closeModal = useCallback(() => {
+    setModalVisible(false);
+  }, []);
+
+  const resetJob = useCallback(async () => {
+    setResetting(true);
+    await resetJobs(jobIds);
+    closeModal();
+
+    setTimeout(() => {
+      refreshJobs();
+    }, RESETTING_JOBS_REFRESH_INTERVAL_MS);
+  }, [jobIds, refreshJobs]);
+
+  if (modalVisible === false || jobIds.length === 0) {
+    return null;
+  }
+
+  return (
+    <EuiModal data-test-subj="mlResetJobConfirmModal" onClose={closeModal}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          <FormattedMessage
+            id="xpack.ml.jobsList.resetJobModal.resetJobsTitle"
+            defaultMessage="Reset {jobsCount, plural, one {{jobId}} other {# jobs}}?"
+            values={{
+              jobsCount: jobIds.length,
+              jobId: jobIds[0],
+            }}
+          />
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <>
+          <OpenJobsWarningCallout jobs={jobs} />
+          {resetting === true ? (
+            <div>
+              <FormattedMessage
+                id="xpack.ml.jobsList.resetJobModal.resettingJobsStatusLabel"
+                defaultMessage="Resetting {jobsCount, plural, one {{jobId}} other {# jobs}}"
+                values={{
+                  jobsCount: jobIds.length,
+                  jobId: jobIds[0],
+                }}
+              />
+              <EuiSpacer />
+              <div style={{ textAlign: 'center' }}>
+                <EuiLoadingSpinner size="l" />
+              </div>
```

Comment:
```suggestion
              <EuiText textAlign="center">
                <EuiLoadingSpinner size="l" />
              </EuiText>
```

Reply:
This whole `resetting` spinner is never shown. I meant to delete the code.

Reply:
Updated in https://github.com/elastic/kibana/pull/108039/commits/06f04dc840fb5a6cf2d98bd62bf8fa7a77f9b747